### PR TITLE
Update to AutoKey ISSUE_TEMPLATE.md

### PR DESCRIPTION
**Pull request by Elliria**
These are my proposed changes to the existing AutoKey issue template. Some of the header syntax was missing and has been fixed and I also edited some of the punctuation and wording in the hopes of making the experience of submitting issues easier or at least a little bit less confusing. Suggestions or improvements are welcome.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -1,50 +1,42 @@
 ## Classification:
-
-(Pick one of: Bug, Crash/Hang/Data Loss, Performance, UI/Usability, Feature (New), Enhancement)
+(Pick one: Bug, Crash/Hang/Data loss, Enhancement, Feature (new), Performance, UI/Usability)
```
Comment (BlueDrink9) [inline .github/ISSUE_TEMPLATE.md:2]: Most markdown flavours won't render headings without a newline after them. Could you verify that github will render something like this correctly?

## My test
No newline

Comment (BlueDrink9) [reply .github/ISSUE_TEMPLATE.md:2]: Hmm, github seems fine with it.

Comment (Elliria) [reply .github/ISSUE_TEMPLATE.md:2]: Yep. You can see the proposed template in action [here](https://github.com/autokey/autokey/blob/c32fb4c59b890288ec6c731aea3a97d27139d826/.github/ISSUE_TEMPLATE.md
).

**Comment by josephj11** [comment]
Looks OK. I know you didn't change it, but I have no idea what the practical difference between a new feature and an enhancement is. I think it should be one or the other. The point is that it's something to add or change, not something to fix.

Too bad markdown doesn't support forms (AFAIK). This would a great application for one. It's somewhat distracting to read issues with all the prompts cluttering them up. Sounds like something that must already be solved.

It is [solved](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms), but it looks there's a bit of a learning curve involved.

Or someone could write an issue text cleaner as a standalone filter. We could copy and paste into it and then replace the issue with the results. Seems doable, but not trivial.

**Comment by BlueDrink9** [comment]
Or just ask users to replace the prompts instead of leaving them in place... Probably a better solution. That's what most projects do

**Comment by Elliria** [comment]
At first I thought that that would be easy to answer, but I can't tell the difference between them, either. This might help:

I looked the labels over and these haven't been used and can possibly be done away with entirely:
- Crash
- Hang
- Data loss
- Feature(new)
- Performance
- UI/Usability

And these have been used in the past and aren't mentioned in the issue template and maybe they should be:
- Development
- Documentation issue
- Installation/Configuration
- Phrase expansion
- Scripting
- User Interface

**Comment by Elliria** [comment]
The YAML form looks glorious. I love the idea of a form that prompts users without inserting any text into the form for them. Check this out: [https://www.youtube.com/watch?v=40r3WhCQL08](https://www.youtube.com/watch?v=40r3WhCQL08)

I'd be happy to help with designing the YAML form, but I'm limited in what I can do. This is my first encounter with YAML, so I'll need to do some learning, although I took a quick look online and it doesn't seem very complicated, at least as far as what we might want it to do.

Also, my knowledge of the back end of AutoKey is limited. I don't know what you need a bug report to provide you at your end. In other words, what it must give you and what's optional.

Last, but not least, I'm not sure if there's a way to test out YAML and see it in action to make certain it does what you want it to do before trying to make it official. If you have a resource for that, I'd love to know about it.

**Comment by Elliria** [comment]
As an aside, I found the MarkDown example on that page that you linked interesting because they used comments to prompt the user. I like the idea of that, since they'd be hidden even if left in. The problem with them, though, would arise when people not familiar with coding come along and have no idea that those are comments, so they put their text into the comments and it never gets seen. Even if it only happened every once in a while, someone would have to be on patrol to check over all bug reports thoroughly to make sure there wasn't any missing information in any of them.

**Comment by Elliria** [comment]
By the way, it looks like I've got a message from GitHub telling me that I'm all set, that my patch branch can be safely deleted and that I can also delete the fork of autokey/autokey in my settings. It's offering me a button and a link to do both of those things. Should I go ahead and wipe those out?

**Comment by josephj11** [comment]
Issue Categories - It might be nice if these corresponded with the assignable issue labels that we use. Both the categories and the labels could be changed to get them to match up. Just a thought - not a recommendation at this point.

Issue reporting requirements - see this [Guide](https://github.com/autokey/autokey-python2/wiki/Problem-Reporting-Guide) (and copy it over to the current branch if you feel motivated - because I never got around to doing it.) It hasn't been updated in quite a while, but I don't think there's anything obvious that needs fixing.

The only thing I know about YAML is what the acronym stands for (and I just looked it up and found I was wrong about even that). I suppose we can look into it together if you want to play with it (I would try to help if you get stuck...)

I poked around for a while, but did not see an application that will render a YAML file so you could test it standalone, but there must be a way to meet such an obvious requirement. I asked about it on [Stack Overflow](https://stackoverflow.com/questions/70079295/how-do-i-test-a-yaml-form-for-problem-reporting-on-a-github-project-page-before). Got one answer so far: Create a dummy GitHub project and install the file there for testing, Copy and install it here once it works. I have an old one lying around and I invited you into it. There's nothing useful there, so you don't have to worry about breaking anything.

I haven't looked at the markdown part, but your concern seems valid. We don't need to make anything more complicated.

Safe to delete - if you look at the line below the title to this issue, it shows that it has been merged into the master branch, so it's in production now. You can do whatever you want with your private repo/copy, but you might want to keep it around for a while in case you want to make further changes. I don't know enough about Git/GitHub to know if it's better to just keep going or to start over with a fresh copy.

**Comment by Elliria** [comment]
I agree on the synchronization of the issue labels with those in the bug report. Should we bring it up in Gitter chat to make sure everyone else has a chance to weigh in on it, too?

I'll let someone else copy over the guide (don't trust myself to do that correctly), but thanks for the link. I borrowed some of its wording for the YAML file I've been working on.

I knew as little as you about YAML and probably had the same incorrect guess on its name as you did (yet another markup language was my guess), but have since done some research. It seems like a relatively straight-forward language, although the documentation for it is all over the place, so you have to go hunting when you want to learn anything fancy.

It looks like your responder on Stack Overflow is right and I see that you created a project for us to mess around in. I cobbled together the **bug.yaml** file and uploaded it to your project, but it's not in the right place. I don't seem to have the power to put it where it belongs, so according to [this YouTube video](https://www.youtube.com/watch?v=40r3WhCQL08), you'll need to create the **.github** folder in the root folder of your project (if it doesn't already exist) and then create the **ISSUE_TEMPLATE** folder inside of it and then put my **bug.yaml** file inside of that. Then we can see if we've got the beginnings of a bug report form that would work for AutoKey.

**Comment by BlueDrink9** [comment]
Yaml is straightforward, but I recommend editing it in browser with github because it's easy to make small syntax mistakes. If you do it on your own fork, you should be able to test it on your own fork too - just try making a new issue on it

**Comment by Elliria** [comment]
Okay that sounds good. I left a copy of it in the wrong place in Joe's repository that he had set up for us to collaborate on and, meanwhile, also put a copy of it into the correct place in my fork of AutoKey just to see if I could get it working.

What's most interesting is that my YAML code passes muster when run through the [YAML Lint](http://www.yamllint.com/) interface, but GitHub is smacking me down hard for getting lots of things wrong, so I'm putting out fires to see if I can get it to work.

If I get a working version and Joe hasn't had a chance to mess with the copy I put into his repository, I'll get rid of that one and replace it with a copy of my working one in the proper place this time (since I figured out how to do that).

**Comment by Elliria** [comment]
Well, I managed to get one working. You can [take a look at it in my temporary fork of AutoKey here](https://github.com/Elliria/autokey-fork/issues). I left in the hundreds of edits I made in case you want to look over what a wrestle that was. It's working well now, though.

Feel free to create one or more issues in there so you can get a feel for what it's like to use those YAML forms. It's not an official repository and is just there temporarily for experimental purposes.

There's an existing issue that I created that we all might want to look over and consider. Oh, and I also copied the working YAML form over to Joe's repository so he's got an exact duplicate of it (and the bug report) to mess around with.

**Comment by josephj11** [comment]
@Elliria So, do I need to fix my repo for the YAML stuff or are you fine using your own fork...?

**Comment by Elliria** [comment]
Both or either. Whatever you like is fine. We both have a working copy of YAML at this point in our repositories, but I've made further changes to my copy (which you're welcome to if you'd like it) and I posted about it in the Gitter chat to get involvement of others, if possible.

**Comment by josephj11** [comment]
Great. Let's continue this discussion elsewhere - like your repo and/or Gitter.

**Pull request merged**